cfe_internal/update/update_policy.cf

```diff
@@ -236,16 +236,18 @@ bundle agent cfe_internal_update_policy_cpv
       action => u_immediate,
       classes => u_if_repaired("validated_updates_ready");
 
-    !am_policy_hub::
+    any::
 
       "$(sys.workdir)$(const.dirsep)modules"
-      comment => "Always update modules files on client side",
+      comment => "Update modules from masterfiles",
       handle => "cfe_internal_update_policy_files_update_modules",
-      copy_from => u_rcp("$(modules_dir_source)", @(update_def.policy_servers)),
+      copy_from => u_cp("$(master_location)$(const.dirsep)modules"),
```

Comment:
It should still copy the package modules from the policy server(s), right?

Reply:
Probably it should first copy modules from the policy server(s), then it should place the modules that are vendored by the mpf.

Reply:
? I thought the package modules will be part of MPF and thus under `/var/cfengine/masterfiles/` and thus updated like everything else.

Reply:
Right. Somewhere else policy copies from `/var/cfengine/masterfiles/` on hub to `/var/cfengine/inputs` on agent. Here we should be copying from `/var/cfengine/inputs/modules` (not masterfiles!) to `/var/cfengine/modules` - and do it on all machines.

@nickanderson : is it okay to do it in two steps or is it better to copy to `/var/cfengine/modules` from `/var/cfengine/masterfiles/modules` at the same time as copying to `/var/cfengine/inputs` from `/var/cfengine/masterfiles` on hub?

Reply:
Two promises are OK I think, they are different targets. or did you mean something else?

Reply:
I mean that if they don't get executed in proper order, there might be an agent run when a policy is updated, but the modules are not there.